File: frontend/src/data/risk-assessment.ts
import type { RiskAssessmentListItem, RiskDashboardData } from "@/types/risk"

type MockDashboardOptions = {
  riskType?: string
  dataSource?: string
}

const BASE_COUNTRY_ENTRIES = [
  {
    code: "US",
    name: "United States",
    score: 82,
    riskLevel: "Low" as const,
    trend: "Stable" as const,
    confidence: "High" as const,
    nextAssessment: "2024-08-05"
  },
  {
    code: "DE",
    name: "Germany",
    score: 76,
    riskLevel: "Low" as const,
    trend: "Improving" as const,
    confidence: "High" as const,
    nextAssessment: "2024-07-22"
  },
  {
    code: "IN",
    name: "India",
    score: 66,
    riskLevel: "Medium" as const,
    trend: "Improving" as const,
    confidence: "Medium" as const,
    nextAssessment: "2024-07-12"
  },
  {
    code: "BR",
    name: "Brazil",
    score: 58,
    riskLevel: "Medium" as const,
    trend: "Deteriorating" as const,
    confidence: "Medium" as const,
    nextAssessment: "2024-06-28"
  },
  {
    code: "ZA",
    name: "South Africa",
    score: 47,
    riskLevel: "High" as const,
    trend: "Deteriorating" as const,
    confidence: "Medium" as const,
    nextAssessment: "2024-06-19"
  },
  {
    code: "RU",
    name: "Russia",
    score: 38,
    riskLevel: "Critical" as const,
    trend: "Deteriorating" as const,
    confidence: "Low" as const,
    nextAssessment: "2024-05-30"
  },
  {
    code: "CN",
    name: "China",
    score: 64,
    riskLevel: "Medium" as const,
    trend: "Stable" as const,
    confidence: "Medium" as const,
    nextAssessment: "2024-07-02"
  }
]

const CATEGORY_TEMPLATES = [
  {
    id: 1,
    name: "Political Stability",
    weight: 20,
    baseline: 78
  },
  {
    id: 2,
    name: "Economic Outlook",
    weight: 20,
    baseline: 74
  },
  {
    id: 3,
    name: "Regulatory Environment",
    weight: 15,
    baseline: 70
  },
  {
    id: 4,
    name: "Corruption Index",
    weight: 10,
    baseline: 68
  },
  {
    id: 5,
    name: "Infrastructure Quality",
    weight: 10,
    baseline: 72
  },
  {
    id: 6,
    name: "Currency Stability",
    weight: 10,
    baseline: 66
  },
  {
    id: 7,
    name: "Trade Relations",
    weight: 10,
    baseline: 69
  },
  {
    id: 8,
    name: "Security Environment",
    weight: 5,
    baseline: 64
  }
]

const OFFLINE_ASSESSMENT_TEMPLATE: RiskAssessmentListItem[] = [
  {
    id: 101,
    title: "Global Political & Economic Heatmap",
    assessmentType: "Comprehensive Risk Assessment",
    status: "Active",
    startDate: "2024-02-01",
    endDate: "2024-05-31",
    nextAssessmentDue: "2024-07-30",
    totalCountries: 18,
    highRiskCountries: 4,
    assignedAssessor: "Sophia Martinez"
  },
  {
    id: 102,
    title: "Strategic Supply Chain Review",
    assessmentType: "Operational Risk Assessment",
    status: "In Review",
    startDate: "2024-01-15",
    endDate: "2024-04-30",
    nextAssessmentDue: "2024-06-15",
    totalCountries: 9,
    highRiskCountries: 2,
    assignedAssessor: "Daniel Carter"
  },
  {
    id: 103,
    title: "Quarterly Emerging Markets Pulse",
    assessmentType: "Economic Risk Assessment",
    status: "Planned",
    startDate: "2024-05-01",
    endDate: "2024-07-31",
    nextAssessmentDue: "2024-08-20",
    totalCountries: 12,
    highRiskCountries: 3,
    assignedAssessor: "Amina Hassan"
  }
]

function buildCountryCategories(multiplier: number) {
  return CATEGORY_TEMPLATES.map((category, index) => {
    const adjustment = Math.sin((multiplier + index) * 1.2) * 8
    const trend = adjustment > 2 ? "Improving" : adjustment < -2 ? "Deteriorating" : "Stable"
    return {
      id: category.id,
      name: category.name,
      score: Math.max(30, Math.min(95, Math.round(category.baseline + adjustment))),
      weight: category.weight,
      aiSuggestion: Math.max(30, Math.min(95, Math.round(category.baseline + adjustment / 2))),
      volatility: Math.abs(Math.round(adjustment)),
      trend
    }
  })
}

function buildSupportingSignals(score: number) {
  const normalized = score / 100
  return {
    economicMomentum: Number((0.4 + normalized * 0.3).toFixed(2)),
    governanceQuality: Number((0.45 + normalized * 0.25).toFixed(2)),
    geopoliticalStability: Number((0.35 + normalized * 0.28).toFixed(2))
  }
}

export function createMockDashboard(options: MockDashboardOptions = {}): RiskDashboardData {
  const riskType = options.riskType ?? "Overall"
  const dataSource = options.dataSource ?? "Combined"
  const generatedAt = new Date().toISOString()

  const countries = BASE_COUNTRY_ENTRIES.map((entry, index) => ({
    code: entry.code,
    name: entry.name,
    score: entry.score,
    riskLevel: entry.riskLevel,
    trend: entry.trend,
    confidence: entry.confidence,
    nextAssessment: entry.nextAssessment
  }))

  const highRiskCount = countries.filter((country) => country.riskLevel === "High" || country.riskLevel === "Critical").length

  const countryDetails = BASE_COUNTRY_ENTRIES.map((entry, index) => {
    const categories = buildCountryCategories(index + 1)
    const impactLevel = entry.riskLevel === "Critical" ? "Critical" : entry.riskLevel === "High" ? "High" : "Medium"
    const probabilityLevel = entry.trend === "Improving" ? "Possible" : entry.trend === "Stable" ? "Likely" : "Almost Certain"

    return {
      id: index + 1,
      countryCode: entry.code,
      countryName: entry.name,
      overallScore: entry.score,
      riskLevel: entry.riskLevel,
      trend: entry.trend,
      confidence: entry.confidence,
      impactLevel,
      probabilityLevel,
      evidence: `${entry.name} ${riskType.toLowerCase()} outlook synthesised from regional indicators and regulatory updates.`,
      comments: `Monitoring ${riskType.toLowerCase()} exposure alongside ${dataSource.toLowerCase()} intelligence feed.`,
      nextAssessmentDue: entry.nextAssessment,
      updateSource: "AI Baseline",
      lastUpdated: generatedAt,
      aiAlerts: entry.riskLevel === "Critical"
        ? ["Escalating sanctions exposure", "High volatility in commodity pricing"]
        : entry.riskLevel === "High"
          ? ["Currency stress testing recommended"]
          : ["Routine monitoring"],
      supportingSignals: buildSupportingSignals(entry.score),
      attachments: [],
      recentChange: entry.trend === "Improving" ? 4 : entry.trend === "Stable" ? 1 : -5,
      categories
    }
  })

  const nextAssessmentDue = countryDetails.reduce<string | undefined>((soonest, country) => {
    if (!country.nextAssessmentDue) return soonest
    if (!soonest) return country.nextAssessmentDue
    return new Date(country.nextAssessmentDue) < new Date(soonest) ? country.nextAssessmentDue : soonest
  }, undefined)

  return {
    generatedAt,
    filters: {
      riskType,
      dataSource
    },
    summary: {
      totalCountries: countries.length,
      highRiskCountries: highRiskCount,
      recentRiskChanges: 5,
      nextAssessmentDue
    },
    map: {
      countries,
      legend: [
        { label: "Low Risk", min: 75, max: 100, color: "#16a34a" },
        { label: "Moderate", min: 55, max: 74, color: "#f59e0b" },
        { label: "Elevated", min: 40, max: 54, color: "#f97316" },
        { label: "Critical", min: 0, max: 39, color: "#ef4444" }
      ]
    },
    aiInsights: {
      trendHighlights: [
        {
          country: "India",
          trend: "Growth indicators improving",
          predictedChange: 6,
          alerts: ["Monitor inflation pressure", "Supply chain stability improving"]
        },
        {
          country: "Brazil",
          trend: "Volatility across commodities",
          predictedChange: -4,
          alerts: ["Political landscape impacting compliance", "Watch currency hedging strategies"]
        },
        {
          country: "Germany",
          trend: "Strong regulatory resilience",
          predictedChange: 3,
          alerts: ["ESG obligations tightening", "Cyber readiness improving"]
        }
      ],
      regionalClusters: {
        "North America": {
          average_score: 81,
          high_risk_ratio: 0.08,
          improving_ratio: 0.62
        },
        Europe: {
          average_score: 73,
          high_risk_ratio: 0.12,
          improving_ratio: 0.58
        },
        "Asia Pacific": {
          average_score: 69,
          high_risk_ratio: 0.18,
          improving_ratio: 0.54
        },
        "Latin America": {
          average_score: 57,
          high_risk_ratio: 0.32,
          improving_ratio: 0.29
        }
      },
      watchlist: [
        {
          country: "Russia",
          riskLevel: "Critical",
          confidence: "Low",
          signals: {
            sanctionsExposure: 0.91,
            commodityShock: 0.88,
            geopoliticalRisk: 0.95
          }
        },
        {
          country: "South Africa",
          riskLevel: "High",
          confidence: "Medium",
          signals: {
            energyReliability: 0.73,
            regulatoryChange: 0.64,
            socialClimate: 0.69
          }
        },
        {
          country: "Brazil",
          riskLevel: "Medium",
          confidence: "Medium",
          signals: {
            currencyVolatility: 0.61,
            fiscalPressure: 0.57,
            complianceAlerts: 0.52
          }
        }
      ]
    },
    countryDetails
  }
}

export function getOfflineAssessments(): RiskAssessmentListItem[] {
  return OFFLINE_ASSESSMENT_TEMPLATE.map((assessment) => ({ ...assessment }))
}

export const MOCK_RISK_ASSESSMENTS = getOfflineAssessments()

File: backend/app/routes/risk_assessment.py
from __future__ import annotations

from datetime import date, datetime, timedelta
from typing import Any, Dict, Iterable, List, Optional

from fastapi import APIRouter, Depends, HTTPException, Query
from pydantic import BaseModel, Field, validator
from sqlalchemy.orm import Session

from database import get_db
from models import (
    ConfidenceLevelEnum,
    CountryRiskCategoryScore,
    CountryRiskProfile,
    DataSourceType,
    RiskAssessment,
    RiskAssessmentStatus,
    RiskTrendEnum,
    UpdateSourceEnum,
)
from app.ai.risk_ai import RiskIntelligenceEngine


router = APIRouter(prefix="/api/risk-assessment", tags=["risk-assessment"])


DEFAULT_CATEGORIES: List[Dict[str, Any]] = [
    {"key": "political_stability", "label": "Political Stability", "weight": 20},
    {"key": "economic_outlook", "label": "Economic Indicators", "weight": 20},
    {"key": "regulatory_index", "label": "Regulatory Environment", "weight": 15},
    {"key": "corruption_index", "label": "Corruption Index", "weight": 10},
    {"key": "infrastructure", "label": "Infrastructure Quality", "weight": 10},
    {"key": "currency_stability", "label": "Currency Stability", "weight": 10},
    {"key": "trade_relations", "label": "Trade Relations", "weight": 10},
    {"key": "security", "label": "Security Environment", "weight": 5},
]


DEFAULT_IMPACT_LEVELS = {
    "Low": "Minimal disruption, manageable within routine operations",
    "Medium": "Noticeable disruption requiring departmental coordination",
    "High": "Significant disruption with executive oversight",
    "Critical": "Severe disruption threatening strategic objectives",
}


DEFAULT_PROBABILITY_LEVELS = {
    "Rare": "Unlikely to occur (<10%)",
    "Possible": "May occur occasionally (10-40%)",
    "Likely": "Expected periodically (40-70%)",
    "Almost Certain": "Anticipated frequently (>70%)",
}


DEFAULT_COUNTRIES = [
    {"code": "DE", "name": "Germany"},
    {"code": "BR", "name": "Brazil"},
    {"code": "SG", "name": "Singapore"},
    {"code": "ZA", "name": "South Africa"},
    {"code": "US", "name": "United States"},
    {"code": "GB", "name": "United Kingdom"},
    {"code": "CN", "name": "China"},
]


class RiskCategoryConfig(BaseModel):
    key: str
    label: str
    weight: float = Field(ge=0, le=100)


class CountrySelection(BaseModel):
    code: str = Field(..., min_length=2, max_length=3)
    name: str


class AssessmentCreatePayload(BaseModel):
    title: str
    assessment_type: str = Field(..., alias="assessmentType")
    countries: List[CountrySelection]
    framework: Optional[str] = None
    scoring_scale: str = Field("1-100", alias="scoringScale")
    assessment_period_start: date = Field(..., alias="assessmentPeriodStart")
    assessment_period_end: date = Field(..., alias="assessmentPeriodEnd")
    update_frequency: str = Field(..., alias="updateFrequency")
    assigned_assessor: str = Field(..., alias="assignedAssessor")
    review_team: List[str] = Field(default_factory=list, alias="reviewTeam")
    categories: List[RiskCategoryConfig] = Field(default_factory=list)
    impact_levels: Dict[str, str] = Field(default_factory=dict, alias="impactLevels")
    probability_levels: Dict[str, str] = Field(default_factory=dict, alias="probabilityLevels")
    data_source: DataSourceType = Field(DataSourceType.COMBINED, alias="dataSource")
    initial_scores: Optional[Dict[str, Dict[str, float]]] = Field(default=None, alias="initialScores")

    @validator("countries")
    def _require_countries(cls, value: List[CountrySelection]) -> List[CountrySelection]:
        if not value:
            raise ValueError("At least one country must be selected for the assessment")
        return value


class CategoryScorePayload(BaseModel):
    name: str
    score: float
    trend: Optional[RiskTrendEnum] = None


class CountryUpdatePayload(BaseModel):
    country_code: str = Field(..., alias="countryCode")
    country_name: Optional[str] = Field(None, alias="countryName")
    overall_score: Optional[float] = Field(None, alias="overallScore")
    trend: Optional[RiskTrendEnum] = None
    confidence: Optional[ConfidenceLevelEnum] = None
    evidence: Optional[str] = None
    comments: Optional[str] = None
    impact_level: Optional[str] = Field(None, alias="impactLevel")
    probability_level: Optional[str] = Field(None, alias="probabilityLevel")
    update_source: UpdateSourceEnum = Field(UpdateSourceEnum.MANUAL, alias="updateSource")
    categories: List[CategoryScorePayload] = Field(default_factory=list)
    attachments: List[Dict[str, Any]] = Field(default_factory=list)
    ai_refresh: bool = Field(False, alias="aiRefresh")


def _engine_for_assessment(assessment: RiskAssessment) -> RiskIntelligenceEngine:
    return RiskIntelligenceEngine(scoring_scale=assessment.scoring_scale)


def _serialize_category(category: CountryRiskCategoryScore) -> Dict[str, Any]:
    return {
        "id": category.id,
        "name": category.name,
        "score": category.score,
        "weight": category.weight,
        "aiSuggestion": category.ai_suggestion,
        "volatility": category.volatility,
        "trend": category.trend.value if category.trend else None,
    }


def _serialize_profile(profile: CountryRiskProfile) -> Dict[str, Any]:
    return {
        "id": profile.id,
        "countryCode": profile.country_code,
        "countryName": profile.country_name,
        "overallScore": profile.overall_score,
        "riskLevel": profile.risk_level,
        "trend": profile.trend.value,
        "confidence": profile.confidence.value,
        "impactLevel": profile.impact_level,
        "probabilityLevel": profile.probability_level,
        "evidence": profile.evidence,
        "comments": profile.comments,
        "nextAssessmentDue": profile.next_assessment_due,
        "updateSource": profile.update_source.value,
        "lastUpdated": profile.last_updated,
        "aiAlerts": profile.ai_alerts or [],
        "supportingSignals": profile.supporting_signals or {},
        "attachments": profile.attachments or [],
        "recentChange": profile.recent_change,
        "categories": [_serialize_category(category) for category in profile.categories],
    }


def _serialize_assessment(assessment: RiskAssessment) -> Dict[str, Any]:
    return {
        "id": assessment.id,
        "title": assessment.title,
        "assessmentType": assessment.assessment_type,
        "framework": assessment.framework,
        "scoringScale": assessment.scoring_scale,
        "updateFrequency": assessment.update_frequency,
        "dataSource": assessment.data_source.value,
        "startDate": assessment.start_date,
        "endDate": assessment.end_date,
        "assignedAssessor": assessment.assigned_assessor,
        "reviewTeam": assessment.review_team or [],
        "status": assessment.status.value,
        "categories": assessment.categories_config,
        "impactLevels": assessment.impact_levels or {},
        "probabilityLevels": assessment.probability_levels or {},
        "aiRecommendations": assessment.ai_recommendations or {},
        "nextAssessmentDue": assessment.next_assessment_due,
        "externalDataSources": assessment.external_data_sources or [],
        "countries": [_serialize_profile(profile) for profile in assessment.countries],
    }


def _ensure_seed_data(session: Session) -> RiskAssessment:
    existing = session.query(RiskAssessment).order_by(RiskAssessment.created_at.desc()).first()
    if existing:
        return existing

    engine = RiskIntelligenceEngine()
    assessment = RiskAssessment(
        title="Global Political & Economic Heatmap",
        assessment_type="Comprehensive Risk Assessment",
        framework="ISO 31000 Risk Management",
        scoring_scale="1-100",
        update_frequency="Quarterly",
        data_source=DataSourceType.COMBINED,
        start_date=datetime.utcnow().date() - timedelta(days=45),
        end_date=datetime.utcnow().date() + timedelta(days=45),
        assigned_assessor="Global Risk Office",
        review_team=["Regulatory Intelligence", "Finance Control", "Security Operations"],
        status=RiskAssessmentStatus.IN_PROGRESS,
        categories_config=DEFAULT_CATEGORIES,
        impact_levels=DEFAULT_IMPACT_LEVELS,
        probability_levels=DEFAULT_PROBABILITY_LEVELS,
        next_assessment_due=datetime.utcnow().date() + timedelta(days=30),
        external_data_sources=[
            {"name": "World Bank Indicators", "lastUpdated": datetime.utcnow() - timedelta(hours=6)},
            {"name": "IMF Country Reports", "lastUpdated": datetime.utcnow() - timedelta(days=1)},
            {"name": "Transparency International", "lastUpdated": datetime.utcnow() - timedelta(days=3)},
        ],
    )
    session.add(assessment)
    session.flush()

    for country in DEFAULT_COUNTRIES:
        intelligence = engine.score_country(
            country_code=country["code"],
            categories=[(cfg["key"], 3.5) for cfg in DEFAULT_CATEGORIES],
            weights={cfg["key"]: cfg["weight"] for cfg in DEFAULT_CATEGORIES},
            scale="1-5",
            assessment_end=assessment.end_date,
        )
        profile = CountryRiskProfile(
            assessment_id=assessment.id,
            country_code=country["code"],
            country_name=country["name"],
            overall_score=intelligence.overall_score,
            risk_level=intelligence.risk_level,
            trend=RiskTrendEnum(intelligence.trend),
            confidence=ConfidenceLevelEnum(intelligence.confidence),
            impact_level="High" if intelligence.risk_level in {"High", "Critical"} else "Medium",
            probability_level="Likely" if intelligence.trend == "Deteriorating" else "Possible",
            evidence="Seeded dataset using AI intelligence model",
            update_source=UpdateSourceEnum.AI,
            next_assessment_due=intelligence.next_assessment,
            ai_alerts=intelligence.ai_alerts,
            supporting_signals=intelligence.supporting_signals,
            recent_change=intelligence.predicted_change,
        )
        session.add(profile)
        session.flush()

        for category in intelligence.category_insights:
            session.add(
                CountryRiskCategoryScore(
                    country_id=profile.id,
                    name=category.name,
                    score=category.score,
                    weight=category.weight,
                    ai_suggestion=category.ai_suggestion,
                    volatility=category.volatility,
                    trend=RiskTrendEnum(intelligence.trend),
                )
            )

    assessment.ai_recommendations = engine.build_ai_dashboard(
        [engine.score_country(country["code"], [(cfg["key"], 3.5) for cfg in DEFAULT_CATEGORIES], {cfg["key"]: cfg["weight"] for cfg in DEFAULT_CATEGORIES}, "1-5", assessment.end_date) for country in DEFAULT_COUNTRIES]
    )
    session.commit()
    session.refresh(assessment)
    return assessment


def _collect_intelligence(assessment: RiskAssessment) -> List[Dict[str, Any]]:
    engine = _engine_for_assessment(assessment)
    insights = []
    for profile in assessment.countries:
        categories = [(category.name, category.score) for category in profile.categories]
        weights = {cfg["key"]: cfg.get("weight", 0) for cfg in assessment.categories_config}
        intelligence = engine.score_country(
            country_code=profile.country_code,
            categories=categories,
            weights=weights,
            scale=assessment.scoring_scale,
            assessment_end=assessment.end_date,
        )
        insights.append({
            "model": intelligence,
            "profile": profile,
        })
    return insights


@router.get("/dashboard")
def dashboard(
    risk_type: str = Query("Overall", alias="riskType"),
    data_source: DataSourceType = Query(DataSourceType.COMBINED, alias="dataSource"),
    session: Session = Depends(get_db),
):
    assessment = _ensure_seed_data(session)
    insights = _collect_intelligence(assessment)
    engine = _engine_for_assessment(assessment)

    profiles = [item["profile"] for item in insights]
    models = [item["model"] for item in insights]

    total_countries = len(profiles)
    high_risk = sum(1 for profile in profiles if profile.risk_level in {"High", "Critical"})
    recent_window = datetime.utcnow() - timedelta(days=14)
    recent_changes = sum(1 for profile in profiles if profile.last_updated >= recent_window)
    upcoming = min((profile.next_assessment_due for profile in profiles if profile.next_assessment_due), default=None)

    return {
        "generatedAt": datetime.utcnow(),
        "filters": {"riskType": risk_type, "dataSource": data_source.value},
        "summary": {
            "totalCountries": total_countries,
            "highRiskCountries": high_risk,
            "recentRiskChanges": recent_changes,
            "nextAssessmentDue": upcoming,
        },
        "map": {
            "countries": [
                {
                    "code": profile.country_code,
                    "name": profile.country_name,
                    "score": profile.overall_score,
                    "riskLevel": profile.risk_level,
                    "trend": profile.trend.value,
                    "confidence": profile.confidence.value,
                    "nextAssessment": profile.next_assessment_due,
                }
                for profile in profiles
            ],
            "legend": [
                {"label": "Low", "min": 0, "max": 25, "color": "#34d399"},
                {"label": "Medium", "min": 26, "max": 50, "color": "#facc15"},
                {"label": "High", "min": 51, "max": 75, "color": "#fb923c"},
                {"label": "Critical", "min": 76, "max": 100, "color": "#f87171"},
            ],
        },
        "aiInsights": engine.build_ai_dashboard([item["model"] for item in insights]),
        "countryDetails": [_serialize_profile(profile) for profile in profiles],
        "riskType": risk_type,
        "dataSource": data_source.value,
    }


@router.get("/assessments")
def list_assessments(session: Session = Depends(get_db)):
    _ensure_seed_data(session)
    assessments = (
        session.query(RiskAssessment)
        .order_by(RiskAssessment.created_at.desc())
        .all()
    )
    response = []
    for assessment in assessments:
        total = len(assessment.countries)
        high_risk = sum(1 for country in assessment.countries if country.risk_level in {"High", "Critical"})
        response.append(
            {
                "id": assessment.id,
                "title": assessment.title,
                "assessmentType": assessment.assessment_type,
                "status": assessment.status.value,
                "startDate": assessment.start_date,
                "endDate": assessment.end_date,
                "nextAssessmentDue": assessment.next_assessment_due,
                "totalCountries": total,
                "highRiskCountries": high_risk,
                "assignedAssessor": assessment.assigned_assessor,
            }
        )
    return response


@router.post("/assessments", status_code=201)
def create_assessment(payload: AssessmentCreatePayload, session: Session = Depends(get_db)):
    categories = payload.categories or [RiskCategoryConfig(**category) for category in DEFAULT_CATEGORIES]
    weights = {category.key: category.weight for category in categories}

    assessment = RiskAssessment(
        title=payload.title,
        assessment_type=payload.assessment_type,
        framework=payload.framework,
        scoring_scale=payload.scoring_scale,
        update_frequency=payload.update_frequency,
        data_source=payload.data_source,
        start_date=payload.assessment_period_start,
        end_date=payload.assessment_period_end,
        assigned_assessor=payload.assigned_assessor,
        review_team=payload.review_team,
        status=RiskAssessmentStatus.SCHEDULED,
        categories_config=[category.model_dump() for category in categories],
        impact_levels=payload.impact_levels or DEFAULT_IMPACT_LEVELS,
        probability_levels=payload.probability_levels or DEFAULT_PROBABILITY_LEVELS,
        next_assessment_due=payload.assessment_period_end,
        external_data_sources=[
            {"name": "World Bank Indicators", "lastUpdated": datetime.utcnow() - timedelta(hours=2)},
            {"name": "Local News Sources", "lastUpdated": datetime.utcnow() - timedelta(hours=1)},
        ],
    )
    session.add(assessment)
    session.flush()

    engine = _engine_for_assessment(assessment)
    intelligence_models = []

    for country in payload.countries:
        initial_scores = payload.initial_scores.get(country.code) if payload.initial_scores else None
        category_scores = []
        for category in categories:
            if initial_scores and category.key in initial_scores:
                base_score = initial_scores[category.key]
            else:
                base_score = 3.5 if payload.scoring_scale != "1-100" else 65
            category_scores.append((category.key, base_score))

        intelligence = engine.score_country(
            country_code=country.code,
            categories=category_scores,
            weights=weights,
            scale=payload.scoring_scale,
            assessment_end=assessment.end_date,
        )
        intelligence_models.append(intelligence)

        profile = CountryRiskProfile(
            assessment_id=assessment.id,
            country_code=country.code.upper(),
            country_name=country.name,
            overall_score=intelligence.overall_score,
            risk_level=intelligence.risk_level,
            trend=RiskTrendEnum(intelligence.trend),
            confidence=ConfidenceLevelEnum(intelligence.confidence),
            impact_level="High" if intelligence.risk_level in {"High", "Critical"} else "Medium",
            probability_level="Likely" if intelligence.trend == "Deteriorating" else "Possible",
            evidence="AI generated baseline using global indicators",
            update_source=UpdateSourceEnum.AI,
            next_assessment_due=intelligence.next_assessment,
            ai_alerts=intelligence.ai_alerts,
            supporting_signals=intelligence.supporting_signals,
            recent_change=intelligence.predicted_change,
        )
        session.add(profile)
        session.flush()

        for category, intelligence_category in zip(categories, intelligence.category_insights):
            session.add(
                CountryRiskCategoryScore(
                    country_id=profile.id,
                    name=category.key,
                    score=intelligence_category.score,
                    weight=intelligence_category.weight,
                    ai_suggestion=intelligence_category.ai_suggestion,
                    volatility=intelligence_category.volatility,
                    trend=RiskTrendEnum(intelligence.trend),
                )
            )

    assessment.ai_recommendations = engine.build_ai_dashboard(intelligence_models)
    session.commit()
    return {"id": assessment.id, "message": "Risk assessment created"}


@router.get("/assessments/{assessment_id}")
def get_assessment(assessment_id: int, session: Session = Depends(get_db)):
    assessment = session.query(RiskAssessment).filter(RiskAssessment.id == assessment_id).first()
    if not assessment:
        raise HTTPException(status_code=404, detail="Assessment not found")
    return _serialize_assessment(assessment)


@router.post("/assessments/{assessment_id}/countries")
def update_country(assessment_id: int, payload: CountryUpdatePayload, session: Session = Depends(get_db)):
    assessment = session.query(RiskAssessment).filter(RiskAssessment.id == assessment_id).first()
    if not assessment:
        raise HTTPException(status_code=404, detail="Assessment not found")

    profile = (
        session.query(CountryRiskProfile)
        .filter(
            CountryRiskProfile.assessment_id == assessment_id,
            CountryRiskProfile.country_code == payload.country_code.upper(),
        )
        .first()
    )
    if not profile:
        profile = CountryRiskProfile(
            assessment_id=assessment_id,
            country_code=payload.country_code.upper(),
            country_name=payload.country_name or payload.country_code.upper(),
            overall_score=0,
            risk_level="Low",
            trend=RiskTrendEnum.STABLE,
            confidence=ConfidenceLevelEnum.MEDIUM,
        )
        session.add(profile)
        session.flush()

    profile.country_name = payload.country_name or profile.country_name
    profile.evidence = payload.evidence or profile.evidence
    profile.comments = payload.comments or profile.comments
    profile.impact_level = payload.impact_level or profile.impact_level
    profile.probability_level = payload.probability_level or profile.probability_level
    profile.update_source = payload.update_source
    profile.attachments = payload.attachments or profile.attachments
    profile.last_updated = datetime.utcnow()

    existing_categories = {category.name: category for category in profile.categories}
    for category_payload in payload.categories:
        category = existing_categories.get(category_payload.name)
        if not category:
            category = CountryRiskCategoryScore(
                country_id=profile.id,
                name=category_payload.name,
                weight=next(
                    (cfg.get("weight") for cfg in assessment.categories_config if cfg["key"] == category_payload.name),
                    0,
                )
                / 100,
            )
            session.add(category)
        category.score = category_payload.score
        category.trend = category_payload.trend or category.trend

    engine = _engine_for_assessment(assessment)
    weights = {cfg["key"]: cfg.get("weight", 0) for cfg in assessment.categories_config}
    intelligence = engine.score_country(
        country_code=profile.country_code,
        categories=[(category.name, category.score) for category in profile.categories],
        weights=weights,
        scale=assessment.scoring_scale,
        assessment_end=assessment.end_date,
    )

    profile.overall_score = intelligence.overall_score
    profile.risk_level = intelligence.risk_level
    profile.trend = RiskTrendEnum(intelligence.trend)
    profile.confidence = ConfidenceLevelEnum(intelligence.confidence)
    profile.next_assessment_due = intelligence.next_assessment
    profile.ai_alerts = intelligence.ai_alerts
    profile.supporting_signals = intelligence.supporting_signals
    profile.recent_change = intelligence.predicted_change

    assessment.ai_recommendations = engine.build_ai_dashboard(
        [
            engine.score_country(
                country.country_code,
                [(category.name, category.score) for category in country.categories],
                weights,
                assessment.scoring_scale,
                assessment.end_date,
            )
            for country in assessment.countries
        ]
    )
    session.commit()
    session.refresh(profile)
    return _serialize_profile(profile)


@router.post("/assessments/{assessment_id}/ai-refresh")
def ai_refresh(assessment_id: int, session: Session = Depends(get_db)):
    assessment = session.query(RiskAssessment).filter(RiskAssessment.id == assessment_id).first()
    if not assessment:
        raise HTTPException(status_code=404, detail="Assessment not found")

    engine = _engine_for_assessment(assessment)
    weights = {cfg["key"]: cfg.get("weight", 0) for cfg in assessment.categories_config}
    models = []
    for profile in assessment.countries:
        intelligence = engine.score_country(
            country_code=profile.country_code,
            categories=[(category.name, category.score) for category in profile.categories],
            weights=weights,
            scale=assessment.scoring_scale,
            assessment_end=assessment.end_date,
        )
        profile.overall_score = intelligence.overall_score
        profile.risk_level = intelligence.risk_level
        profile.trend = RiskTrendEnum(intelligence.trend)
        profile.confidence = ConfidenceLevelEnum(intelligence.confidence)
        profile.next_assessment_due = intelligence.next_assessment
        profile.ai_alerts = intelligence.ai_alerts
        profile.supporting_signals = intelligence.supporting_signals
        profile.recent_change = intelligence.predicted_change
        models.append(intelligence)

    assessment.ai_recommendations = engine.build_ai_dashboard(models)
    session.commit()
    session.refresh(assessment)
    return {
        "updatedAt": datetime.utcnow(),
        "aiRecommendations": assessment.ai_recommendations,
        "countries": [_serialize_profile(profile) for profile in assessment.countries],
    }
